tpm/algorithm/algorithm.go

```diff
@@ -0,0 +1,98 @@
+package algorithm
+
+import (
+	"encoding/json"
+	"slices"
+
+	"github.com/google/go-tpm/legacy/tpm2"
+)
+
+var (
+	algs map[tpm2.Algorithm]string
+)
+
+type Algorithm tpm2.Algorithm
+
+func (a Algorithm) String() string {
+	return algs[tpm2.Algorithm(int(a))]
+}
+
+func (a Algorithm) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.String())
+}
+
+type AlgorithmSlice struct {
+	algs []Algorithm
+}
```

Comment:
This possibly can be made a type alias. Depending on usage, could be named just `Algorithms`.

Reply:
I replaced this type and `Supports`/`Contains` with `SupportsAlgorithms` in the `tpm` package.